test/fixtures/parser/snowflake/snowflake_set_session_variable.yml

```diff
@@ -0,0 +1,150 @@
+file:
+- statement:
+    set_session_variable_statement:
+      keyword: SET
+      parameter: var1
+      comparison_operator: '='
+      expression:
+        literal: '10'
+- statement_terminator: ;
+- statement:
+    set_session_variable_statement:
+      keyword: SET
+      parameter: var1
+      comparison_operator: '='
+      expression:
+        literal: 'TRUE'
+- statement_terminator: ;
+- statement:
+    set_session_variable_statement:
+      keyword: SET
+      parameter: var1
+      comparison_operator: '='
+      expression:
+        literal: "'example'"
+- statement_terminator: ;
+- statement:
+    set_session_variable_statement:
+      keyword: SET
+      parameter: var1
+      comparison_operator: '='
+      expression:
+        bare_function: CURRENT_TIMESTAMP
+- statement_terminator: ;
+- statement:
+    set_session_variable_statement:
+      keyword: SET
+      parameter: var1
+      comparison_operator: '='
+      expression:
+        variable_reference: $var2
```

Comment:
* Running `python test/generate_parse_fixture_yml.py` is able to parse the variable correctly as a `variable_reference` as defined in https://github.com/sqlfluff/sqlfluff/pull/1001/files#diff-a8da7f7dc8f0e448950f6d5c2cd6cd5590026ea6da32cdc8f233e037bea947ecR112